### PR TITLE
restic: add s3CredentialsFile option

### PR DESCRIPTION
**Pull request by jerith666**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/backup/restic.nix [hunk 1]
```diff
@@ -14,7 +14,15 @@ with lib;
             Read the repository password from a file.
           '';
           example = "/etc/nixos/restic-password";
+        };
 
+        s3CredentialsFile = mkOption {
+          type = types.str;
```
Comment (Mic92) [inline nixos/modules/services/backup/restic.nix]: I suppose it should have a default value: 

```
type = with types; nullOr str;
default = null;
```

File: nixos/modules/services/backup/restic.nix [hunk 2]
```diff
@@ -134,6 +142,8 @@ with lib;
             Type = "oneshot";
             ExecStart = "${resticCmd} backup ${concatStringsSep " " backup.extraBackupArgs} ${concatStringsSep " " backup.paths}";
             User = backup.user;
+          } // optionalAttrs (backup.s3CredentialsFile != "") {
```
Comment (Mic92) [inline nixos/modules/services/backup/restic.nix]: With the new default it can be compared with null instead of empty string.

**Comment by jerith666** [comment]
Hm ... am I missing a `default = null;` in the `mkOption{}`?

**Comment by jerith666** [comment]
yep, I am -- sorry!  fixed in #44332.

**Comment by jerith666** [comment]
was just too focused on that use case -- okay w/ me if you want to rename it.

**Pull request merged**